### PR TITLE
KOGITO-371: KOGITO-389: [DMN Designer] Marshalling location of nodes inside a decision service

### PR DESCRIPTION
**Pull request by manstis**
See:-
- https://issues.jboss.org/browse/KOGITO-371
- https://issues.jboss.org/browse/KOGITO-389

One PR, two commits (not to be squashed) one for each JIRA (389 follows on from 371).

Unit tests to be covered by https://issues.jboss.org/browse/KOGITO-406

### CODE REVIEW

**Comment by manstis** [comment]
@jomarko This PR is good to review.

**Comment by manstis** [comment]
@jomarko The Sonar review was triggered following an additional line-break added to the class.

If you insist I will fix some of the issues - but I was not about to perform a full refactor of the class. I'd rather remove the additional line break so the class is not scanned by Sonar! WDYT?

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java [hunk 1]
```diff
@@ -179,45 +170,61 @@ private void doTransformation(final String xml,
...
+        final Graph graph = diagram.getGraph();
         if (Objects.isNull(graph)) {
             return;
         }
 
-        final DMN12MarshallCallback jsCallback = xml -> {
-            String breakpoint = xml;
-            if (!breakpoint.startsWith("<?xml version=\"1.0\" ?>")) {
-                breakpoint = "<?xml version=\"1.0\" ?>" + breakpoint;
+        final DMN12MarshallCallback jsCallback = r -> {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java]: does `r` stands for `result`? if yes please update.

**Comment by jomarko** [comment]
@manstis regarding sonar, I do not insist on 80% boundary in PRs (you might notice), however ignoring code smells like this one is bad. Could we report a jira?

Screenshot from 2019-10-30 13-15-08

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshallerStandalone.java [hunk 2]
```diff
@@ -423,18 +410,16 @@ public Graph unmarshall(final Metadata metadata,
...
-                            .stream()
-                            .filter(componentWidths -> componentWidths.getDmnElementRef().getLocalPart().equals(es.getKey()))
-                            .findFirst()
-                            .ifPresent(componentWidths -> {
-                                final List<Double> widths = es.getValue().getComponentWidths();
-                                widths.clear();
-                                widths.addAll(componentWidths.getWidths());
-                            });
-                });
+                hasComponentWidthsMap.forEach((uuid, hasComponentWidths) -> componentsWidthsExtension
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/DMNMarshallerStandalone.java]: @jomarko Grid columns (so cell) width.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/DMNClientDiagramServiceImpl.java]: Yes, updated.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/pom.xml [hunk 3]
```diff
@@ -183,11 +183,6 @@
       <artifactId>slf4j-api</artifactId>
     </dependency>
 
-    <!-- TODO {manstis} This is temporary until DMNMarshaller is fully implemented -->
-    <dependency>
-      <groupId>org.jboss.errai</groupId>
-      <artifactId>errai-marshalling</artifactId>
-    </dependency>
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/pom.xml]: I am the latest person in kiegorup team who would block such change, however now mvn dependency analyze says:
`errai-marshalling` is a `Used undeclared dependency`

Is it fine?

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/pom.xml]: Reinstated.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@jomarko This PR can be re-reviewed.. it seems OK to merge.

Test Coverage is to be handled by https://issues.jboss.org/browse/KOGITO-405 and https://issues.jboss.org/browse/KOGITO-406

**Comment by manstis** [comment]
@jomarko Hi, I've fixed your last reported [problem](https://github.com/kiegroup/kie-wb-common/pull/2974#pullrequestreview-313194437) :-)

The problem seemed to be pasting the _content_ of a third-party DMN file into one that had been created in `-webapp-kogito-testing` where the third-party file name contained a space. Pasting the _content_ of one file into another created with the webapp is the _normal_ way to test different files.

I managed to replicate the issue with `Loan Pre-Qualification.dmn` and can confirm the latest commit resolves the issue. I tested that `-webapp-kogito-testing` and `-webapp-kogito-runtime` work correctly. The last commit contains no changes that could impact either Business Central or `-webapp-standalone`.

**Comment by manstis** [comment]
@jomarko When you get a chance this PR can be re-tested.. once this is has been merged we _should_ be able to use `master` to build `kogito-tooling` plugins and perform subsequent testing in VSCode.

**Comment by jomarko** [comment]
@manstis I am not testing third part files in that way, I just 
- put file into `stunner/diagrams`
- git add `stunner/diagrams/diagram.dmn`
- git commit -m "brutal commit"
- git push origin master

The issue I reported can be still spotted using steps above. The file I used is [here](https://issues.jboss.org/secure/attachment/12457387/Drawing%201.dmn)

**Comment by manstis** [comment]
Thanks @jomarko I'll take a look.

**Comment by manstis** [comment]
Thank-you for the re-producer @jomarko.

The issue related to spaces in the filename, so I now encode it and `Drawing 1.dmn` works OK now (all be it that the diagram itself is rather devoid of names etc but the DMN text doesn't have any either so I assume that's how it's meant to be).

**Comment by manstis** [comment]
Reminder that code coverage is to be handled by:
- https://issues.jboss.org/browse/KOGITO-405
- https://issues.jboss.org/browse/KOGITO-406

**Pull request merged**